cli/bpmetadata/validate_test.go

```diff
@@ -0,0 +1,58 @@
+package bpmetadata
+
+import (
+	"path"
+	"testing"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	yamlTestDirPath = "../testdata/bpmetadata/schema"
+)
+
+func TestValidateMetadata(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr bool
+	}{
+		{
+			name:    "empty metadata",
+			path:    "empty-metadata.yaml",
+			wantErr: true,
+		},
+		{
+			name:    "valid metadata",
+			path:    "valid-metadata.yaml",
+			wantErr: false,
+		},
+		{
+			name:    "invalid metadata - source missing",
+			path:    "invalid-metadata.yaml",
+			wantErr: true,
+		},
+		{
+			name: "valid enums for QuotaType",
+			path: "valid-metadata-w-enum.yaml",
+		},
+		{
+			name:    "invalid enums for QuotaResourceType",
+			path:    "invalid-metadata-w-enum.yaml",
+			wantErr: true,
+		},
+	}
+
+	// load schema from the binary
+	s := gojsonschema.NewReferenceLoader("file://schema/bpmetadataschema.json")
```

Comment:
Rather than using a static pre generated file, if we generate the metadata as part of the test, that would help us catch any issues when we change the schema.

Reply:
Got it. I can do this as a follow-up because I'll need to restructure the packages since call generate right now would create a cyclic dependency.

Reply:
#1449